docs/pages/repo/docs/core-concepts/monorepos/skipping-tasks.mdx

```diff
@@ -0,0 +1,62 @@
+---
+title: Ignoring tasks
+description: Never do the same work twice.
+---
+
+import Callout from '../../../../../components/Callout'
+
+# Ignoring Tasks
+
+[Build caching](/repo/docs/core-concepts/caching) can dramatically speed up your build tasks - but you can do even better by using `npx turbo-ignore`. If your workspace is unaffected by your code changes, you can completely skip executing a task altogether.
+
+## Using `turbo-ignore`
+
+To skip unaffected work with `npx turbo-ignore`, ensure that your git history is available and run the command.
+
+By default, `turbo-ignore` will run the **build** task in the current working directory. Under the hood, `turbo-ignore` compares the output of `--dry=json` from your previous commit to your current commit.
+
+Here's an example of the command that will be built and ran:
+
+```bash
+npx turbo run build --filter=@example/web...[3c8387ffd98b751305fe3f0284befdd00cbd4610] --dry=json
+```
+
+Note that this command does not _execute_ the build task. Instead, it checks if a **dry run** of a build is different from the **dry run** of the task from the previous commit.
+
+If `turbo-ignore` finds that the task can be skipped, it will exit the process with a `0` code. If changes have been found, the process will exist with `1`.
+
+<Callout type="info">On Vercel, the previously deployed SHA will be used instead of the previous commit.</Callout>
+
+## Customizing behavior
+
+If you'd like to change the default behavior, there are a few flags available:
+
+- `--workspace`: Specifies the workspace for the filter command used by `turbo-ignore`. Defaults to the `name` field of the `package.json` in the directory where `turbo-ignore` is invoked.
+- `--task`: Specifies the task for the command that `turbo-ignore` will invoke. Defaults to `build`.
+- `--directory`: Specify a directory where `turbo-ignore` will run.
+- `--fallback`: Specify a ref/head to compare against.
+
+## Example
+
+Let's say you want to skip the unit tests for your `web` workspace when there aren't any changes to your `web` application or its package dependencies. In practice, hitting the cache with a `>>> FULL TURBO` will speed things up quite a bit but there will still be time spent installing `npm` dependencies, creating upcoming CI infrastructure, or doing other types of work that can take awhile.
+
+Ideally, we would do a quick check to see if any of that work needs to happen in the first place.
+
+After we've checked out the repo but **before** any other work, we can take a few seconds to check that our `web` tests have changed since the previous commit.
+
+```bash
+npx turbo-ignore --workspace=web --task=test
+```
+
+This command will:
+
+- filter for the `web` workspace
+- create the `dry` output for your `test` task at the _current_ commit
+- create the `dry` output for your `test` task at the _previous_ commit
+- compare the results of the two `dry` outputs
```

Comment:
This actually isn't true. It's just one dry run - the filter tells turbo (in this example) I'm only concerned about web, from now until commit x. And turbo determines what workspaces need to be built, and returns that in the dry-run, and then we parse that response in `turbo ignore` to determine which packages have changed. 

Here's an example - started from create-turbo, with a single change to ui:

Command is: 
`npx turbo run build --filter=web...[HEAD^] --dry=json`

The result is:
```
{
  "packages": [
    "web"
  ],
  "tasks": [
    {
      "taskId": "eslint-config-custom#build",
      "task": "build",
      "package": "eslint-config-custom",
      "hash": "ecc5fe5ea423e6b9",
      "cacheState": {
        "local": false,
        "remote": false
      },
      "command": "\u003cNONEXISTENT\u003e",
      "outputs": [
        ".next/**",
        "dist/**"
      ],
      "excludedOutputs": null,
      "logFile": "packages/eslint-config-custom/.turbo/turbo-build.log",
      "directory": "packages/eslint-config-custom",
      "dependencies": [],
      "dependents": [
        "ui#build",
        "web#build"
      ],
      "resolvedTaskDefinition": {
        "outputs": [
          ".next/**",
          "dist/**"
        ],
        "cache": true,
        "dependsOn": [
          "^build"
        ],
        "inputs": [],
        "outputMode": "full",
        "env": [],
        "persistent": false
      }
    },
    {
      "taskId": "tsconfig#build",
      "task": "build",
      "package": "tsconfig",
      "hash": "8f782633436459be",
      "cacheState": {
        "local": false,
        "remote": false
      },
      "command": "\u003cNONEXISTENT\u003e",
      "outputs": [
        ".next/**",
        "dist/**"
      ],
      "excludedOutputs": null,
      "logFile": "packages/tsconfig/.turbo/turbo-build.log",
      "directory": "packages/tsconfig",
      "dependencies": [],
      "dependents": [
        "ui#build",
        "web#build"
      ],
      "resolvedTaskDefinition": {
        "outputs": [
          ".next/**",
          "dist/**"
        ],
        "cache": true,
        "dependsOn": [
          "^build"
        ],
        "inputs": [],
        "outputMode": "full",
        "env": [],
        "persistent": false
      }
    },
    {
      "taskId": "ui#build",
      "task": "build",
      "package": "ui",
      "hash": "3018d1a0af40066a",
      "cacheState": {
        "local": false,
        "remote": false
      },
      "command": "\u003cNONEXISTENT\u003e",
      "outputs": [
        ".next/**",
        "dist/**"
      ],
      "excludedOutputs": null,
      "logFile": "packages/ui/.turbo/turbo-build.log",
      "directory": "packages/ui",
      "dependencies": [
        "tsconfig#build",
        "eslint-config-custom#build"
      ],
      "dependents": [
        "web#build"
      ],
      "resolvedTaskDefinition": {
        "outputs": [
          ".next/**",
          "dist/**"
        ],
        "cache": true,
        "dependsOn": [
          "^build"
        ],
        "inputs": [],
        "outputMode": "full",
        "env": [],
        "persistent": false
      }
    },
    {
      "taskId": "web#build",
      "task": "build",
      "package": "web",
      "hash": "f01d4428937286ac",
      "cacheState": {
        "local": false,
        "remote": false
      },
      "command": "next build",
      "outputs": [
        ".next/**",
        "dist/**"
      ],
      "excludedOutputs": null,
      "logFile": "apps/web/.turbo/turbo-build.log",
      "directory": "apps/web",
      "dependencies": [
        "ui#build",
        "tsconfig#build",
        "eslint-config-custom#build"
      ],
      "dependents": [],
      "resolvedTaskDefinition": {
        "outputs": [
          ".next/**",
          "dist/**"
        ],
        "cache": true,
        "dependsOn": [
          "^build"
        ],
        "inputs": [],
        "outputMode": "full",
        "env": [],
        "persistent": false
      }
    }
  ]
}
```

We parse this output here: https://github.com/vercel/turbo/blob/main/packages/turbo-ignore/src/ignore.ts#L105
to see what turbo returns in the packages array to determine what has changed.